plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XProjectManager.java

```diff
@@ -198,11 +195,13 @@ protected XBuildResult doBuild(Set<URI> dirtyFiles, Set<URI> deletedFiles,
 		XBuildResult result = incrementalBuilder.build(request);
 
 		projectStateHolder.updateProjectState(request, result);
+		if (doGenerate && result.getAffectedResources().isEmpty()) {
```

Comment:
I think the condition `persistedProjectStateOutdated` was inverse to this guard here

Reply:
Maybe a `!` before `….isEmpty()` is missing?

Reply:
Yes, I think so